File: src/swig/perl/play.pl
#!/usr/bin/env perl

# Import required modules
use mlt;

# Not sure why the mlt::Factory.init method fails...
mlt::mlt_factory_init( undef );

# Establish the MLT profile
$profile = new mlt::Profile( undef );

# Create the producer
$p = new mlt::Producer( $profile, $ARGV[0] );

if ( $p->is_valid( ) )
{
	# Loop the video
	$p->set( "eof", "loop" );

	# Create the consumer
	$c = new mlt::FilteredConsumer( $profile, "sdl" );

	# Turn of the default rescaling
	$c->set( "rescale", "none" );

	# Connect the producer to the consumer
	$c->connect( $p );
	
	$e = $c->setup_wait_for( "consumer-stopped" );

	# Start the consumer
	$c->start;

	# Wait until the user stops the consumer
	$c->wait_for( $e );

	$e = undef;
	$c = undef;
	$p = undef;
}
else
{
	print "Unable to open $ARGV[0]\n";
}

mlt::mlt_factory_close( );

File: src/swig/nodejs/play.js
#!/usr/bin/env node

const path = require('path')

const mlt = require('./node-gyp-build/Release/mlt.node')

const {
    Factory,
    Profile,
    Producer,
    Consumer
} = mlt;

const videoFile = process.argv[2]
console.log(videoFile)

Factory.init('')

let profile = new Profile('hdv_720_25p')
let producer = new Producer(profile, 'avformat', videoFile)


if (!producer.is_valid()) {
    console.log(`Unable to open producer`)
    process.exit(1)
}

console.log('get_fps', producer.get_fps())

let consumer = new Consumer(profile, 'sdl')

consumer.connect(producer)
consumer.start()

check()

function check() {
    console.log('is_stopped', consumer.is_stopped())
    if (consumer.is_stopped()) {
        consumer.stop()
        Factory.close()
        return
    }
    setTimeout(check, 1000)
}


